eventstore/views.py

```diff
@@ -20,6 +27,89 @@
     PrebirthRegistrationSerializer,
     PublicRegistrationSerializer,
 )
+from ndoh_hub.utils import validate_signature
+
+
+class MessagesViewSet(GenericViewSet):
+    def create(self, request):
+        validate_signature(request)
+        try:
+            webhook_type = request.META.get("HTTP_X_TURN_HOOK_SUBSCRIPTION", "whatsapp")
+            if webhook_type == "whatsapp":
+                for inbound in request.data.get("messages", []):
+                    data = {
+                        "context": inbound.get("context"),
+                        "errors": inbound.get("errors"),
+                        "audio": inbound.get("audio"),
+                        "document": inbound.get("document"),
+                        "image": inbound.get("image"),
+                        "system": inbound.get("system"),
+                        "location": inbound.get("location"),
+                        "text": inbound.get("text"),
+                        "video": inbound.get("video"),
+                        "voice": inbound.get("voice"),
+                    }
+                    Messages.objects.create(
+                        id=inbound["id"],
+                        contact_id=inbound["from"],
+                        type=inbound["type"],
+                        data=data,
+                        message_direction=Messages.INBOUND,
+                        created_by=request.user.username,
+                        recipient_type=inbound["recipient_type"],
+                        timestamp=datetime.fromtimestamp(int(inbound["timestamp"])),
+                    )
+
+                for statuses in request.data.get("statuses", []):
+                    data = {"errors": statuses.get("errors")}
```

Comment:
Same here for rather using `pop` to get the required fields, and then putting the remaining data in `data`